### PR TITLE
set socket timeout before connect

### PR DESCRIPTION
**Pull request by mrname**

### CODE REVIEW

File: src/clamd/__init__.py [hunk 1]
```diff
@@ -63,8 +63,8 @@ def _init_socket(self):
         """
         try:
             self.clamd_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            self.clamd_socket.connect((self.host, self.port))
```
Comment (RishiRamraj) [inline src/clamd/__init__.py:66]: Docs please :)

**Review by Apakottur — APPROVED** [review]
:pretzel:

File: src/clamd/__init__.py (hunk 1 above)
Comment (mrname) [reply src/clamd/__init__.py:66]: Can you explain what you think should be documented? The fact that the existing code calls `connect` before `settimeout` seems more like an oversight than anything else.

Comment (RishiRamraj) [reply src/clamd/__init__.py:66]: Why does the order matter?

**Review by pawk — APPROVED** [review]

File: src/clamd/__init__.py (hunk 1 above)
Comment (mrname) [reply src/clamd/__init__.py:66]: Well I mean... it is explicitly stated in the python docs: https://docs.python.org/3/library/socket.html#timeouts-and-the-connect-method

You think we should comment in this repo as well?

Comment (RishiRamraj) [reply src/clamd/__init__.py:66]: Your call :) Usually when something trips me up, I write a comment about it.

**Pull request merged**